packages/appsync-emulator-serverless/__test__/lambdaRunner.test.js

```diff
@@ -0,0 +1,82 @@
+const { run } = require('../lambdaRunner');
+
+describe('lambdaRunner', () => {
+  describe('run', () => {
+    describe('callback', () => {
+      test('The lambda throws an error', async () => {
+        const lambda = (event, context, callback) => {
+          callback('error', null);
+        };
+        const context = {};
+        const payload = {};
+
+        run({ lambda, context, payload }, err => {
+          expect(err).toMatch('error');
+        });
+      });
+      test('The lambda returns output', async () => {
+        const lambda = (event, context, callback) => {
+          callback(null, true);
+        };
+        const context = {};
+        const payload = {};
+
+        run({ lambda, context, payload }, (err, output) => {
+          expect(output).toBe(true);
+        });
+      });
+    });
+
+    describe('async', () => {
+      test('The lambda throws an error', async () => {
+        const lambda = async () => {
+          throw new Error('error');
+        };
+        const context = {};
+        const payload = {};
+
+        try {
+          await run({ lambda, context, payload });
+        } catch (e) {
+          expect(e.message).toMatch('error');
+        }
+      });
+      test('The lambda returns output', async () => {
+        const lambda = async () => true;
+        const context = {};
+        const payload = {};
+
+        const output = await run({ lambda, context, payload });
+        expect(output).toBe(true);
+      });
+    });
+
+    describe('promise', () => {
+      test('The lambda throws an error', async () => {
+        const lambda = () =>
+          new Promise((_, reject) => {
+            reject(new Error());
+          });
+        const context = {};
+        const payload = {};
+
+        try {
+          await run({ lambda, context, payload });
+        } catch (e) {
+          expect(e).toBeInstanceOf(Error);
+        }
+      });
+      test('The lambda returns output', async () => {
+        const lambda = () =>
+          new Promise(resolve => {
+            resolve(true);
+          });
+        const context = {};
+        const payload = {};
+
+        const output = await run({ lambda, context, payload });
+        expect(output).toBe(true);
+      });
+    });
+  });
+});
```

Comment:
Great!

Two nitpicks:

1.  Do you mind changing `test` to `it` so that unit tests are consistent throughout the repo?
2.  I do not think we need to distinguish between `async` and `promise` lambda methods as `async` is equivalent to returning a promise.

Reply:
There is a little difference between async function and returning promise function.

- Async function is an instance of `AsyncFunction` and return promise.
- Returning promise function is an `Function` and return promise.

It's possible to change how to check which async pattern is used. This distinguishment would help when you improve lambdaRunner.